src/components/form/file-upload/file-upload.tsx

```diff
@@ -0,0 +1,184 @@
+        return size <= maxFileSizeUpload && maxFileSizeUpload
+            ? setState((prevState) => ({ ...prevState, isValidFileSize: true }))
+            : setState((prevState) => ({ ...prevState, isValidFileSize: false }));
+    };
+
+    return (
+        <div>
+            <section
+                className={clsx(
+                    'flex',
+                    'flex-col',
+                    'justify-center',
+                    'items-center',
+                    'h-[200px]',
+                    'w-full',
+                    'p-xs',
+                    'text-white',
+                    'rounded-s',
+                    'border-2 border-secondary-200',
+                    isDragIsOver ? 'bg-secondary-200' : 'bg-secondary-100',
+                )}
+                onDragOver={handleDragOver}
+                onDragLeave={handleDragLeave}
+                onDrop={handleDrop}
+            >
+                <IconUpload
+                    size={IconSize.L}
+                    className={`mb-xs ${
+                        isValidFileType && isValidFileSize ? 'fill-secondary-500' : 'fill-error'
+                    }`}
+                />
+                <Label
+                    text={label}
+                    size={LabelSize.XL}
+                    className={`mb-xs ${
+                        isValidFileType && isValidFileSize ? 'text-secondary-500' : 'text-error'
+                    }`}
+                />
+                <Paragraph
+                    text={labelFileSize}
+                    size={ParagraphSize.M}
+                    className={`${
+                        isValidFileType && isValidFileSize ? 'text-secondary-500' : 'text-error'
+                    }`}
+                />
+            </section>
+            <input
+                className="hidden"
+                type="file"
+                ref={inputReference}
+                id={fileInputId}
+                multiple={false}
+                onChange={handleChange}
+            />
+            <section className="mb-xl mt-s">
+                {labelButton && (
+                    <Button
+                        size={ButtonSize.M}
+                        variant={Variant.SPECIAL}
+                        onClick={onFileUpload}
+                        label={labelButton}
+                        Icon={Icon}
+                        className="w-full"
+                    />
+                )}
+            </section>
+            <section className="flex w-full flex-row justify-between gap-x-s">
+                <Button
+                    size={ButtonSize.M}
+                    variant={Variant.SECONDARY}
+                    onClick={() => {}}
+                    label={labelCancelButton}
+                    Icon={IconActionLeft}
+                    className="basis-1/2"
+                />
+                <Button
+                    size={ButtonSize.M}
+                    onClick={() => {}}
```

Comment:
Add a new prop in FileUploadProps and pass it to this function here, for example `onSave: () => void`